model/project_parser_s3.go

```diff
@@ -80,8 +80,12 @@ func (s *ParserProjectS3Storage) UpsertOne(ctx context.Context, pp *ParserProjec
 	if err != nil {
 		return errors.Wrap(err, "getting config")
 	}
+	isDegradedMode := !config.ServiceFlags.CPUDegradedModeDisabled
 	maxSize := config.TaskLimits.MaxParserProjectSize
-	if maxSize > 0 && parserProjectLen > maxSize {
+	if isDegradedMode {
+		maxSize = config.TaskLimits.MaxDegradedModeParserProjectSize
+	}
+	if maxSize > 0 && parserProjectLen > maxSize*1024*1024 {
 		return errors.Errorf("parser project exceeds the system limit (%v > %v bytes).", parserProjectLen, maxSize)
```

Comment:
nit: can we clarify whether it's the CPU degraded mode or normal service limit here? Otherwise users might be confused if it changes when we switch to degraded mode.